gems/sorbet-runtime/lib/types/private/methods/signature.rb

```diff
@@ -197,17 +189,31 @@ def each_args_value_type(args)
 
     begin
       it = 0
-      while it < args_length
-        yield arg_types[it][0], args[it], arg_types[it][1]
+
+      # Process given pre-rest args. When there are no rest args,
+      # this is just the given number of args.
+      while it < [args_length, @arg_types.length].min
```

Comment:
You might want to avoid using an array literal here—pretty sure that's going to be another allocation.

If `@arg_types.length` never changes, maybe something like this:

```ruby
while it < args_length && it < arg_types_length
  # ...
end
```